sklearn/decomposition/_pca.py

```diff
@@ -465,13 +465,9 @@ def _fit(self, X):
         """Dispatch to the right submethod depending on the chosen solver."""
         xp, is_array_api_compliant = get_namespace(X)
 
-        # Raise an error for sparse input and unsupported svd_solver
+        # Automatically select "arpack" solver if the input is sparse
```

Comment:
This is not the right change. We need to keep this safe guard here.

We need to change the code below where we defined `_fit_svd_solver`. If `"auto"` and only in this case, we default to `"arpack"`. So we should have a new branch in the `if` statement:

```python
        # Handle svd_solver
        self._fit_svd_solver = self.svd_solver
        if self._fit_svd_solver == "auto":
            # Small problem or n_components == 'mle', just call full PCA
            if issparse(X):
                self._fit_svd_solver = "arpack"
            elif max(X.shape) <= 500 or n_components == "mle":
                self._fit_svd_solver = "full"
            elif 1 <= n_components < 0.8 * min(X.shape):
                self._fit_svd_solver = "randomized"
            # This is also the case of n_components in (0,1)
            else:
                self._fit_svd_solver = "full"
```